File: app/src/main/java/com/arman/queuetube/model/viewholders/BaseViewHolder.java
package com.arman.queuetube.model.viewholders;

import android.view.View;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

public abstract class BaseViewHolder<E> extends RecyclerView.ViewHolder {

    protected E item;

    public BaseViewHolder(@NonNull View itemView) {
        super(itemView);
    }

    public void bind(E item) {
        this.item = item;
    }

}

File: app/src/main/java/com/arman/queuetube/model/viewholders/BaseViewHolder.kt
package com.arman.queuetube.model.viewholders

import android.view.View
import androidx.recyclerview.widget.RecyclerView

abstract class BaseViewHolder<E>(itemView: View) : RecyclerView.ViewHolder(itemView) {

    var item: E? = null
        protected set

    fun bind(item: E) {
        this.item = item
    }

    open fun unbind() {

    }

}
